inspirehep/modules/orcid/tasks.py

```diff
@@ -0,0 +1,125 @@
+                authors = get_orcid_valid_authors(sender)
+                for author in authors:
+                    token, author_orcid = get_authors_credentials(author)
+                    if not InspireOrcidRecords.query.filter_by(orcid=author_orcid, record_id=int(sender.get('control_number'))).all():
+                        put_code = api.add_record(
+                            author_orcid, token, 'work', orcid_json)
+                        x = InspireOrcidRecords(
+                            orcid=author_orcid,
+                            record_id=sender.get('control_number'),
+                            put_code=put_code)
+                        with db.session.begin_nested():
+                            db.session.add(x)
+                    else:
+                        put_code = InspireOrcidRecords.query.filter_by(
+                            orcid=author_orcid, record_id=int(
+                                sender.get('control_number'))).first().put_code
+                        api.update_record(author_orcid, token,
+                                          'work', orcid_json, str(put_code))
+            except RequestException as e:
+                print(e.response.text)
+                print sender['control_number']
+        except (KeyError, AttributeError) as e:
+            exc_type, exc_obj, exc_tb = sys.exc_info()
+            fname = os.path.split(exc_tb.tb_frame.f_code.co_filename)[1]
+            print(exc_type, fname, exc_tb.tb_lineno)
+
+
+def get_authors_credentials(author):
+    """Returns the orcid-id and the orcid-token for a specific author (if available)."""
+    author_orcid = ''
+    author = author['_source']
+    for orcid_id in author['ids']:
+        if orcid_id['type'] == 'ORCID':
+            author_orcid = orcid_id['value']
+    raw_user = UserIdentity.query.filter_by(
+        id=author_orcid, method='orcid').first()
+    user = RemoteAccount.query.get(raw_user.id_user)
+    token = user.tokens[0].access_token
+    return (token, author_orcid)
+
+
+def get_orcid_valid_authors(sender):
+    """ Returns all the valid author-records from a hep-record.
+    A valid author-rerord is one that has 'curated_relation' = True
+    and contains an orcid id.
+    """
+    valid_authors = []
+    authors = sender['authors']
+    for author in authors:
+        if author['curated_relation']:
+            valid_authors.append(author['record']['$ref'])
+    es_query = {
+        "filter": {
+            "bool": {
+                "must": [
+                    {"terms": {
+                        "self.$ref": valid_authors
+                    }}, {"match": {
+                        "ids.type": "ORCID"
+                    }}
+                ]
+            }
+        }
+    }
+    authors = current_search_client.search(
+        index='records-authors',
+        doc_type='authors',
+        body=es_query
+    )['hits']['hits']
+    if authors:
+        return authors
+    else:
+        return []
```

Comment:
Do we need this? Isn't `authors` going to be an empty list anyway? `return authors` should be enough then.